addons/website/static/src/scss/website.scss

```diff
@@ -1460,6 +1460,16 @@ header {
             vertical-align: middle;
         }
     }
+
+    .navbar-collapse.show > &:last-child {
```

Comment:
@robinlej I don't think this is robust enough: e.g. in the "Contact" header template, the rule won't be applied on mobile because the last element is a section (hidden on mobile).

![15 0-rev-ls-scroll--role](https://github.com/odoo/odoo/assets/25032876/b4e73c4f-e736-46be-b0cd-850e512babae)

Maybe we can just target the language selector regardless of its position and limit this fix to the templates where the language selector triggers the scroll behaviour :thinking:… What do you think?

Reply:
> limit this fix to the templates where the language selector triggers the scroll behaviour

That's what I tried at first (in `website_templates.xml`), but the templates causing an issue have a common template between desktop and mobile, meaning that if I set `position-static` on the dropdown, it's also applied on desktop (and since there aren't breakpoint classes for positions, I can't do that either). I wanted to avoid useless JS, which is why I tried tweaking the CSS.
(If you meant to select the templates through CSS, I don't think there are ids, classes or attributes that would differentiate the headers, are there?)

Note that this seems to work, but that's a weird one (and it uses `:has()`, I know it's a hot topic :eyes:), basically checking whether all (if any) following siblings are hidden:
```css
.navbar-collapse.show > &:not(:has(~ :not(.d-none))) {
    .dropdown-toggle {
        display:block;
    }

    .dropdown-menu {
        position: static;
    }
}
```

Reply:
I ended up with a class targeting every header that is concerned, and a media breakpoint on that class.